js/auth/api.js

```diff
@@ -0,0 +1,51 @@
+import { BASE_URL, SIGN_IN_ENDPOINT, CHECK_EMAIL_ENDPOINT, SIGN_UP_ENDPOINT } from './constants.js';
+
+/**
+ * 로그인 시도 api
+ * @param {string} email - 사용자의 이메일 주소
+ * @param {string} password - 사용자의 비밀번호
+ * @returns {Promise<Object>} - 로그인 요청의 결과를 반환
+ */
+export async function signIn(email, password) {
+  const response = await fetch(`${BASE_URL}${SIGN_IN_ENDPOINT}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password }),
+  });
+  return response.json();
+}
+
+/**
+ * 이메일 중복체크 api
+ * @param {string} email - 체크할 이메일 주소
+ * @returns {Promise<Object>} - 이메일 중복 체크 요청의 결과를 반환
+ */
+export async function checkEmail(email) {
+  const response = await fetch(`${BASE_URL}${CHECK_EMAIL_ENDPOINT}`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email }),
+  });
+  return response.json();
+}
+
+/**
+ * 회원가입 api
+ * @param {string} email - 사용자가 사용할 이메일 주소
+ * @param {string} password - 사용자가 설정한 비밀번호
+ * @returns {Promise<Object>} - 회원가입 요청의 결과를 반환
+ */
+export async function signUp(email, password) {
```

Comment:
파라미터를 받을때, 객체로 받는게 좋더라구요 저는
물론 요렇게 2개까진 큰 무리가 없지만, 3개이상이 된다면, 순서에 관계없이 조금더 명시적으로 파라미터를 받고, 넘겨줄수 있어요!
```suggestion
export async function signUp({email, password}) {
// 사용되는 곳에선 signUp({email:'asd@naver.com', password:'1234'}
```